torchsynth/module.py

```diff
@@ -541,22 +541,23 @@ def _forward(self, audio_in: Signal, control_in: Signal) -> Signal:
 
 class Noise(SynthModule):
     """
-    Generates white noise that is the same length as the buffer
+    Generates white noise that is the same length as the buffer.
+
+    A batch size number of noise samples are pre-computed for performance.
```

Comment:
Hmm okay :\ I still would prefer a prime so that things cycle a bit

Reply:
I messed around with this a bit. I found a clean and fast way to do it -- but things get weird as soon as we introduce the deterministic batch retrieval aspect. When we call synth(batch_idx) we need to make sure that the correct cycle of the noise is returned. Would need to spend a bit more time to think that through and implement in a way that doesn't slow things down too much. I'm still pretty convinced that is won't be a perceptually relevant thing. But, let's file an issue for it if you're okay with that.

Reply:
I think this is okay for now.

Reply:
Actually we do need one fix. Create DEFAULT_BATCH_SIZE = 64 and always use that.

We want that if someone is doing batch_size 1024 for speed, they get *identical* results. Ya feel me?

Reply:
I meant, always use DEFAULT_BATCH_SIZE for white noise. And explain it's for reproducibility

Reply:
Might even consider making a test that 2 x batchsize=64 == 1 x batchsize 128 :)

Reply:
Yes totally get this. I think we need to consider how parameters are going to be sampled then too, b/c those need to be updated as well to allow for larger batch sizes